src/main/scala/edu/ie3/simona/model/thermal/ThermalHouse.scala

```diff
@@ -113,23 +124,231 @@ final case class ThermalHouse(
+    * If we already simulated a tick within the current simulation time hour, it
+    * returns None. Else a sequence of the hours since the tick of state will be
+    * returned.
     *
-    * In edge cases, i.e. within the tolerance margin of target temperatures,
-    * the temperature difference can be negative. For these cases we set the
-    * temperature difference to zero, resulting in an energy demand of 0 kWh.
+    * @param tick
+    *   The actual simulation tick.
+    * @param simulationTime
+    *   The actual time of the simulation.
+    * @param state
+    *   The state of the heat pump.
+    * @return
+    *   A sequence of the absolute hour in simulation time. Or None, if we
+    *   already simulated a tick within the actual hour.
+    */
+
+  def checkIfNeedToDetermineDomesticHotWaterDemand(
+      tick: Long,
+      simulationTime: ZonedDateTime,
+      state: HpState,
+  ): Option[Seq[Int]] = {
+    if tick == 0 then {
+      Some(Seq(simulationTime.getHour))
+    } else {
+      val timeDiffSeconds = tick - state.tick
+      val lastStateTime = simulationTime.minusSeconds(timeDiffSeconds)
+      if timeDiffSeconds > 3600 then {
+        val hoursDiff = (timeDiffSeconds / 3600).toInt
+
+        Some(
+          (1 to hoursDiff + 1).map(hour => (lastStateTime.getHour + hour) % 24)
+        )
+
+      } else if simulationTime.getHour != lastStateTime.getHour then {
+        Some(Seq(simulationTime.getHour))
+      } else {
+        None
+      }
+    }
+  }
+
+  /** Calculate the energy demand for warm water at the instance in question.
+    *
+    * @param hoursWaterDemandToDetermine
+    *   Sequence of hours for which the energy demand should be determined.
+    * @param thermalHouseState
+    *   Actual state, that is valid for this model.
+    * @return
+    *   The needed energy for hot domestic water for the questioned hours.
+    */
+
+  def energyDemandDomesticHotWater(
+      hoursWaterDemandToDetermine: Option[Seq[Int]],
+      thermalHouseState: Option[ThermalHouseState],
+  ): ThermalEnergyDemand = {
+    val totalEnergyDemand = hoursWaterDemandToDetermine match {
+      case Some(hours) =>
+        hours.foldLeft(zeroKWh) { (accumulator, hour) =>
+          accumulator + calculateThermalEnergyOfWaterDemand(hour)
+        }
+      case None =>
+        zeroKWh
+    }
+    ThermalEnergyDemand(totalEnergyDemand, totalEnergyDemand)
+  }
+
+  /** Calculate the energy demand for warm water for some specific hour at the
+    * instance in question.
+    *
+    * @param hour
+    *   The hour for that the energy demand should be determined.
+    * @return
+    *   The needed energy for domestic hot water in the questioned hour.
+    */
+
+  private def calculateThermalEnergyOfWaterDemand(hour: Int): Energy = {
+    val waterDemand =
+      waterDemandOfHour(hour, houseInhabitants, housingType)
+    thermalEnergyDemandWater(waterDemand, Celsius(10d), Celsius(55d))
```

Comment:
Why did you hardcode these values? Are that constants? If not maybe we should make them configurable via a `HotWaterStorageConfig` and set these values as defaults.

Reply:
These are based on VDI 2067 Blatt 12 - I don't see the necessity to make these configurable. On stochastic basis temperature preferece of people will be in that range.
I will add the reference in the code.